air/src/stack/mod.rs

```diff
@@ -92,6 +128,36 @@ pub fn enforce_unique_constraints<E: FieldElement>(
     constraint_offset
 }
 
+/// Enforces generic constraints of all the stack operations.
+pub fn enforce_generic_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &op_flags::OpFlags<E>,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // enforces the first element of the stack gets copied over onto the next frame.
+    result[constraint_offset] =
+        op_flag.no_change_0() * are_equal(frame.stack_item(0), frame.stack_item_next(0));
+
+    constraint_offset += 1;
+
+    // enforces the second element of the stack gets copied over onto the next frame.
+    result[constraint_offset] =
+        op_flag.no_change_1() * are_equal(frame.stack_item(1), frame.stack_item_next(1));
+
+    constraint_offset += 1;
+
+    // enforces all the stack items post the second elements get copied over onto the next frame.
+    for i in 2..16 {
+        result[constraint_offset] =
+            op_flag.no_change_after_1() * are_equal(frame.stack_item(i), frame.stack_item_next(i));
```

Comment:
I've handled it using a for loop in the `enforce_generic_function`. This might not be optimal as we might have to update the range or introduce similar loops while introducing new operations. It's better if we handle it in utils.rs in the stack module as you have mentioned in the PR #370. Let me know how do you want to handle it.
Also I was wondering if we should have similar independent generic functions for both left and right shift stack operations?